### ISSUE TITLE
Integrate tidy's output from Travis in GitHub's diff view

### ISSUE THREAD

**Issue by nox** [servo/servo#7480]
Make @highfive or another bot comment on GitHub on the lines reported by Travis's run of `./mach test-tidy`.

**Comment by jdm** [comment]
I don't think this even needs reviewable integration; it can leave comments on the lines of code in the commit view, and reviewable integrates those automatically.

**Comment by nox** [comment]
@jdm Reworded.

**Comment by JoshTheGoldfish** [comment]
Seems like fun, I'll check this one out.

**Comment by jdm** [comment]
There are a few components to this - given a [JSON notification](https://github.com/jdm/highfive/issues/16#issuecomment-129521608) that links to a Travis CI page, retrieve the [relevant content](https://travis-ci.org/servo/servo/builds/74856035) from the build, then use the [github API](https://developer.github.com/v3/pulls/comments/) to leave comments on the right lines.

**Comment by jdm** [comment]
You can find API inspiration at https://github.com/jdm/highfive/blob/master/newpr.py

**Comment by frewsxcv** [comment]
Upon every push, should all the tidy comments be removed?

**Comment by nox** [comment]
New identical ones shouldn't be pushed again, but I don't think it's useful to remove the older ones, they disappear anyway from the diff.

**Comment by JoshTheGoldfish** [comment]
@jdm, I noticed you wrote your tool in Python. However, I'd prefer to use Ruby, as Github and Travis-CI both provide Ruby bindings for their APIs. Is anyone opposed to this?

**Comment by Ms2ger** [comment]
I don't think it would be a good idea for us to gain a dependency on Ruby.

**Comment by nox** [comment]
https://github.com/irqed/octokit.py
https://pypi.python.org/pypi/TravisPy/

**Comment by JoshTheGoldfish** [comment]
@Ms2ger, @nox, No problem, thanks for the info. I'll give it a shot in Python.

**Comment by nox** [comment]
We should find a contributor to do this.

**Comment by mrobinson** [comment]
Going to reopen this with an updated description that doesn't reference Travis.
